src/devices/Ft4222/Ft4222Spi.cs

```diff
@@ -129,85 +134,53 @@ public Ft4222Spi(SpiConnectionSettings settings)
+        /// <inheritdoc/>
         public override void Read(Span<byte> buffer)
         {
-            byte[] readBuff = new byte[buffer.Length];
-            ushort readBytes = 0;
-            var ftStatus = FtFunction.FT4222_SPIMaster_SingleRead(_ftHandle, readBuff, (ushort)buffer.Length, ref readBytes, true);
+            ushort readBytes;
+            var ftStatus = FtFunction.FT4222_SPIMaster_SingleRead(_ftHandle, out MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, out readBytes, true);
             if (ftStatus != FtStatus.Ok)
                 throw new IOException($"{nameof(Read)} failed to read, error: {ftStatus}");
-
-            readBuff.CopyTo(buffer);
         }
 
-        /// <summary>
-        /// Reads a byte from the SPI device.
-        /// </summary>
-        /// <returns>A byte read from the SPI device.</returns>
+        /// <inheritdoc/>
         public override byte ReadByte()
         {
-            byte[] toRead = new byte[1];
+            Span<byte> toRead = stackalloc byte[1];
             Read(toRead);
             return toRead[1];
         }
 
-        /// <summary>
-        /// Writes and reads data from the SPI device.
-        /// </summary>
-        /// <param name="writeBuffer">The buffer that contains the data to be written to the SPI device.</param>
-        /// <param name="readBuffer">The buffer to read the data from the SPI device.</param>
+        /// <inheritdoc/>
         public override void TransferFullDuplex(ReadOnlySpan<byte> writeBuffer, Span<byte> readBuffer)
         {
-            byte[] readBuff = new byte[readBuffer.Length];
-            ushort readBytes = 0;
-            var ftStatus = FtFunction.FT4222_SPIMaster_SingleReadWrite(_ftHandle, readBuff, writeBuffer.ToArray(), (ushort)writeBuffer.Length, ref readBytes, true);
+            ushort readBytes;
+            var ftStatus = FtFunction.FT4222_SPIMaster_SingleReadWrite(_ftHandle, out MemoryMarshal.GetReference(readBuffer), in MemoryMarshal.GetReference(writeBuffer), (ushort)writeBuffer.Length, out readBytes, true);
             if (ftStatus != FtStatus.Ok)
                 throw new IOException($"{nameof(TransferFullDuplex)} failed to do a full duplex transfer, error: {ftStatus}");
-
-            readBuff.CopyTo(readBuffer);
         }
 
-        /// <summary>
-        /// Writes data to the SPI device.
-        /// </summary>
-        /// <param name="buffer">
-        /// The buffer that contains the data to be written to the SPI device.
-        /// </param>
+        /// <inheritdoc/>
         public override void Write(ReadOnlySpan<byte> buffer)
         {
-            ushort writeBytes = 0;
-            var ftStatus = FtFunction.FT4222_SPIMaster_SingleWrite(_ftHandle, buffer.ToArray(), (ushort)buffer.Length, ref writeBytes, true);
+            ushort writeBytes;
+            var ftStatus = FtFunction.FT4222_SPIMaster_SingleWrite(_ftHandle, in MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, out writeBytes, true);
```

Comment:
use name: `bytesWritten` which we use everywhere in the framework (if needed also modify the interop with similar conventions)